### PR TITLE
Initial SSR refactor

### PR DESCRIPTION
**Pull request by pngwn**
## Description

This is the initial part of #8795.

Some light refactoring to make it easier to create both SSR and SPA versions of gradio.

We still need the SPA version so we can fallback to it when node isn't available and also so we can continue to support the embeds.

This PR does the following:

- introduces `@gradio/core`. This contains mosrt of the inner workings of the gradio frontend, exactly as they were. But instead of being an entire app, these parts are now exported for consumption elsewhere.
- introduces `@gradio/spa`. This consumes `@gradio/core` and is the current gradio frontend and what we use to build the actual app currently.
- Refactors the gradio python and gradio lite packages and builds. Instead of `@gradio/lite` being a special build of `@gradio/app` (or `@gradio/core` now). It now has its own dedicated build step + config . All build commands and test commands are contained within the `@gradio/lite` packages. This has simplified the configs + build processes of both `@gradio/spa` and `@gradio/lite`.
- `@gradio/app` is now empty (i'll add the boilerplate in this PR) but this will be the location of the SSR/ node version of the gradio 'app'. I wanted to use 'app' because this more accurately reflects the nature of its contents.

### CODE REVIEW

**Comment by abidlabs** [comment]
Demos look good to me @pngwn! You can ignore the failing `Model3D` demo, as that is also failing on `main`. 

Should be good to merge once js tests pass

**Review by abidlabs — APPROVED** [review]

File: js/wasm/src/worker-proxy.ts [hunk 1]
```diff
@@ -38,7 +38,7 @@ export class WorkerProxy extends EventTarget {
 
 		console.debug("WorkerProxy.constructor(): Create a new worker.");
 		// Loading a worker here relies on Vite's support for WebWorkers (https://vitejs.dev/guide/features.html#web-workers),
-		// assuming that this module is imported from the Gradio frontend (`@gradio/app`), which is bundled with Vite.
+		// assuming that this module is imported from the Gradio frontend (`@gradio/spa`), which is bundled with Vite.
```
Comment (whitphx) [inline js/wasm/src/worker-proxy.ts:41]:
```suggestion
		// assuming that this module is imported from the Gradio frontend (`@gradio/lite`), which is bundled with Vite.
```

File: scripts/build_lite.sh [hunk 2]
```diff
@@ -8,4 +8,4 @@ pnpm_required
 rm -rf gradio/_frontend_code
 rm -rf gradio/templates/*
 
-pnpm --filter @gradio/app build:lite
+pnpm --filter @gradio/spa build
```
Comment (whitphx) [inline scripts/build_lite.sh:11]:
```suggestion
pnpm --filter @gradio/lite build
```

File: =23.2
Comment (whitphx) [inline =23.2:1]: This file should be deleted.

**Review by whitphx — APPROVED** [review]
Thanks! I really like the split spa/lite packages and build configs.
I just reviewed code around `lite` and put nit comments but it overall LGTM.

**Pull request merged**